### PR TITLE
A real overseer feature

### PR DESCRIPTION
**Pull request by montekki**
To test:
```
cargo run --release --features="real-overseer"  -- --tmp --chain rococo-local --validator --alice --force-authoring
```

### CODE REVIEW

**Review by ordian — APPROVED** [review]

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ mod grandpa_support;
 mod client;
 
 use grandpa::{self, FinalityProofProvider as GrandpaFinalityProofProvider};
-#[cfg(feature = "full-node")]
+#[cfg(all(feature = "full-node"))]
```
Comment (bkchr) [inline node/service/src/lib.rs:26]:
```suggestion
#[cfg(feature = "full-node")]
```

**Review by bkchr — APPROVED** [review]

**Comment by rphmeier** [comment]
bot merge

**Comment by joelamouche** [comment]
Hi, so I'm trying to learn more about Overseer and why I need to pass `--features=real-overseer` when testing my parachain (moonbeam).
What happens if I **don't** pass it?
Thanks in advance

**Comment by rphmeier** [comment]
If you don't pass `--features=real-overseer` then all parachains consensus components are replaced with dummy versions that don't do anything.

**Comment by joelamouche** [comment]
So block won't be produced on the parachains?

**Pull request merged**